modules/plugins/languages/ruby.nix

```diff
@@ -0,0 +1,154 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}: let
+  inherit (builtins) attrNames;
+  inherit (lib.options) mkEnableOption mkOption;
+  inherit (lib.modules) mkIf mkMerge;
+  inherit (lib.nvim.types) mkGrammarOption;
+  inherit (lib.types) either listOf package str enum;
+  inherit (lib.nvim.languages) diagnosticsToLua;
+
+  cfg = config.vim.languages.ruby;
+
+  defaultServer = "rubyserver";
+  servers = {
+    rubyserver = {
+      package = pkgs.rubyPackages.solargraph;
+      lspConfig = ''
+        lspconfig.solargraph.setup {
+          capabilities = capabilities,
+          on_attach = attach_keymaps,
+          flags = {
+            debounce_text_changes = 150,
+          },
+          cmd = { "${pkgs.solargraph}/bin/solargraph", "stdio" }
+        }
+      '';
+    };
+  };
+
+  # testing
+
+  defaultFormat = "rubocop";
+  formats = {
+    rubocop = {
+      package = pkgs.rubyPackages.rubocop;
+      nullConfig = ''
+
+        local conditional = function(fn)
+          local utils = require("null-ls.utils").make_conditional_utils()
+          return fn(utils)
+        end
+
+        table.insert(
+          ls_sources,
+          null_ls.builtins.formatting.rubocop.with({
+            command="${pkgs.bundler}/bin/bundle",
+            args = vim.list_extend(
+              {"exec", "rubocop", "-a" },
+              null_ls.builtins.formatting.rubocop._opts.args
+            ),
+          })
+        )
+      '';
+    };
+  };
+
+  defaultDiagnosticsProvider = ["rubocop"];
+  diagnosticsProviders = {
+    rubocop = {
+      package = pkgs.rubyPackages.rubocop;
+      nullConfig = pkg: ''
+        table.insert(
+          ls_sources,
+          null_ls.builtins.diagnostics.rubocop.with({
+            command = "${lib.getExe pkg}",
+          })
+        )
+      '';
+    };
+  };
+in {
+  options.vim.languages.ruby = {
+    enable = mkEnableOption "Ruby language support";
+
+    treesitter = {
+      enable = mkEnableOption "Ruby treesitter" // {default = config.vim.languages.enableTreesitter;};
+      package = mkGrammarOption pkgs "ruby";
+    };
+
+    lsp = {
+      enable = mkEnableOption "Ruby LSP support" // {default = config.vim.languages.enableLSP;};
+
+      server = mkOption {
+        type = enum (attrNames servers);
+        default = defaultServer;
+        description = "Ruby LSP server to use";
+      };
+
+      package = mkOption {
+        description = "Ruby LSP server package, or the command to run as a list of strings";
+        type = either package (listOf str);
+        default = servers.${cfg.lsp.server}.package;
+        example = ''TODO'';
+      };
```

Comment:
Please make sure the mkOption arguments are arranged by their likeliness to change. A good rule of thumb is 
```nix
mkOption {
	type = ...;
	default = ...;
	example = ...;
  description = ...;
}
```

Also, if the example is outside the scope of this PR you may omit it for the time being.